Allows you to toggle off Byond/Discord/URLs in OOC

GeneriedJenelle opened the pull request:
Useful if you're getting hit by spam attacks. Just a config option. Disables byond, discord or regular URLs if you really need to.

GeneriedJenelle committed:
Allows you to toggle off Byond/Discord/URLs in OOC

Useful if you're getting hit by spam attacks. Just a config option.

Spookerton commented on code/controllers/configuration.dm:
```diff
@@ -195,6 +195,10 @@ var/list/gamemode_cache = list()
+	
+	var/allow_byond_links = 0
+	var/allow_discord_links = 0
+	var/allow_url_links = 0					// honestly if I were you i'd leave this one off, only use in dire situations
```
comment should be in the config file, not code

GeneriedJenelle replied:
@Spookerton Hey Spookerton, meant to put this a draft PR. I'm currently occupied as I am having an issue locating your sanitize_and_communicate proc references in ooc.dm to add the code to OOC. I can't find it on github search either, do you know where it is?

Currently porting this, which is drastically different: https://github.com/PolarisSS13/Polaris/pull/7206/files

Edit: Found it.

Spookerton reviewed (dismissed):
- the config options are unused; missed some staging?
- needs an example/config.txt edit

GeneriedJenelle committed:
Update configuration.dm

GeneriedJenelle committed:
adds to ooc

GeneriedJenelle committed:
adds to looc

GeneriedJenelle committed:
Adds to example config

GeneriedJenelle committed:
Adds discord url to config

Might be useful for any discord links you want to add to your interface later, this is for reference only now.

GeneriedJenelle committed:
Update configuration.dm

GeneriedJenelle committed:
adds exception to LOOC that your own discord url can be posted

I realize a lot of people share the server's discord url in OOC, adding exceptions. Admins can now freely post whatever they like.

GeneriedJenelle committed:
Update ooc.dm

GeneriedJenelle committed:
Update looc.dm

GeneriedJenelle committed:
adds discordurl setting to example.txt

GeneriedJenelle commented:
Should be good for now, I'll check back later to see what travis says.

Spookerton commented:
@GeneriedJenelle Looks fine. Squash pleez.

PsiOmegaDelta commented on code/datums/communication/ooc.dm:
```diff
@@ -16,11 +16,19 @@
+		if((findtext(message, "http://") || findtext(message, "https://")) && !config.allow_url_links)
+			to_chat(C, "<B>Posting links in OOC is not allowed.</B>")
+			log_and_message_admins("has attempted to advertise in [name]: [message]")
+			return FALSE
```
This looks like straight copy paste of the block above

PsiOmegaDelta reviewed (changes requested):
Generalize the big blocks into a shared proc

HeyBanditoz commented on code/datums/communication/looc.dm:
```diff
@@ -15,7 +15,21 @@
+			to_chat(C, "<B>Advertising other servers is not allowed.</B>")
+			log_and_message_admins("has attempted to advertise in [name]: [message]")
+			return FALSE
+		if(findtext(message, "discord.gg") && !config.allow_discord_links)
```
Invites can be `https://discord.com/invite/[...]`

GeneriedJenelle commented:
I have a cleaner version of this, will PR soon.
